system/HTTP/OutgoingRequestInterface.php

```diff
@@ -0,0 +1,94 @@
+<?php
+
+/**
+ * This file is part of CodeIgniter 4 framework.
+ *
+ * (c) CodeIgniter Foundation <admin@codeigniter.com>
+ *
+ * For the full copyright and license information, please view
+ * the LICENSE file that was distributed with this source code.
+ */
+
+namespace CodeIgniter\HTTP;
+
+use InvalidArgumentException;
+
+/**
+ * Representation of an outgoing, client-side request.
+ *
+ * Corresponds to Psr7\RequestInterface.
+ */
+interface OutgoingRequestInterface extends MessageInterface
```

Comment:
Let's talk about why OutgoingRequest isn't a request first, but I'm also not sure if we need to interface this. It feels pretty bad to create a new interface with a deprecation in it 😢